CONTRIBUTING.md

```diff
@@ -0,0 +1,178 @@
+# Contributing to SmartFix
+
+Thank you for your interest in contributing to SmartFix! This guide will help you get set up for development.
+
+## 🚀 Quick Start
+
+### 1. Clone and Setup
+
+```bash
+git clone <repository-url>
+cd contrast-ai-smartfix-action
+
+# Install git hooks for automatic linting
+./setup-hooks.sh
+```
+
+### 2. Install Dependencies
+
+```bash
+# Install Python dependencies (creates .venv and installs packages)
+./test/run_tests.sh --skip-install  # Will prompt to install uv if needed
+
+# Or install manually
+pip install -r src/requirements.txt
+pip install flake8  # For linting (if not already installed by setup-hooks.sh)
+```
+
+### 3. Verify Setup
+
+```bash
+# Run tests to ensure everything works
+./test/run_tests.sh
+
+# Run linting to check code quality
+./.git/hooks/pre-push
+```
+
+## 🔧 Development Workflow
+
+### Code Quality & Linting
+
+We use automated linting to maintain code quality with **single source of truth** approach:
+
+- **Local Development**: Git hooks automatically run linting checks
+- **CI/CD**: Uses the same hook scripts to ensure consistency
+
+#### Git Hooks (Automatic)
+- **Pre-commit**: Cleans trailing whitespace
+- **Pre-push**: Runs Python linting with flake8
+
+#### Manual Linting
+```bash
+# Run the same linting that CI uses
+./.git/hooks/pre-push
+
+# Or run flake8 directly (but hook is preferred for consistency)
+flake8 src/ test/
```

Comment:
just a thought...

in other repos we use a `Makefile` and add these recipes or commands to it so it is just 1 command that initiates the specific task. example: `make test` would run this command